src/clusterfuzz/_internal/bot/tasks/utasks/analyze_task.py

```diff
@@ -256,11 +257,22 @@ def handle_noncrash(output):
     testcase.status = 'Unreproducible, retrying'
     testcase.put()
 
+    events.emit(
+        events.TestcaseRejectionEvent(
+            source='handle_noncrash',
+            testcase=testcase,
+            rejection_reason='analyze_flake_on_first_attempt'))
+
     tasks.add_task('analyze', output.uworker_input.testcase_id,
                    output.uworker_input.job_type)
     return
   testcase_upload_metadata = testcase_utils.get_testcase_upload_metadata(
       output.uworker_input.testcase_id)
+  events.emit(
+      events.TestcaseRejectionEvent(
+          source='handle_noncrash',
+          testcase=testcase,
+          rejection_reason='analyze_no_repro'))
   data_handler.mark_invalid_uploaded_testcase(
       testcase, testcase_upload_metadata, 'Unreproducible')
```

Comment:
Should we emit the event after the "logic execution" of the testcase (e.g., marking it as invalid)? @vitorguidi @ViniciustCosta @jonathanmetzman 

```suggestion
  rejection_event = (
      events.TestcaseRejectionEvent(
          source='handle_noncrash',
          testcase=testcase,
          rejection_reason='analyze_no_repro'))
  data_handler.mark_invalid_uploaded_testcase(
      testcase, testcase_upload_metadata, 'Unreproducible')
  events.emit(rejection_event)
```

Reply:
As we were discussing offline, I think this is a better way to do it, since we create the event before making changes to the testcase metadata, to assert that we have access to the data we need for the event, but only emit it after actually executing the method.